tests/test_compiler.py

```diff
@@ -353,6 +353,24 @@ def inner(j):
         assert circuit(5, 6) == 30  # 5 * 6
         assert circuit(4, 7) == 28  # 4 * 7
 
+    def test_while_loop_python_fallback(self):
+        """Test that qml.while_loop fallsback to
+        Python without qjit"""
+
+        def f(n, m):
+            @qml.while_loop(lambda i, _: i < n)
+            def outer(i, sm):
+                @qml.while_loop(lambda j: j < m)
+                def inner(j):
```

Comment:
Might be worth testing the queuing behavior here as well.

Reply:
Do you mean have a QNode/operation example as well?

Reply:
just including some operations, yes.

Reply:
Have added a test here: https://github.com/PennyLaneAI/pennylane/pull/6014/files/ee2532242b9d6da4f6e1c37125519e9fecf94a5b..f833dd33c64aeb8a2589b53b8ea0e3c7f8e2d509